### PR TITLE
Test fix: add longer wait for agent startup for converge tests

### LINKED ISSUE #8204: Test Failure: LogStash::Agent Agent execute options when `config.reload.automatic` is set to`FALSE` and successfully load the config converge only once
Jenkins test failure running compatibility tests on OpenSuse for version 6.x:

```
17:43:03 Failures:
17:43:03 
17:43:03   1) LogStash::Agent Agent execute options when `config.reload.automatic` is set to`FALSE` and successfully load the config converge only once
17:43:03      Failure/Error: expect(source_loader.fetch_count).to eq(1)
17:43:03 
17:43:03        expected: 1
17:43:03             got: 0
17:43:03 
17:43:03        (compared using ==)
17:43:03      # /var/lib/jenkins/workspace/elastic+logstash+6.x+multijob-unix-compatibility/os/opensuse/logstash-core/spec/logstash/agent/converge_spec.rb:108:in `block in (root)'
17:43:03      # /var/lib/jenkins/workspace/elastic+logstash+6.x+multijob-unix-compatibility/os/opensuse/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /var/lib/jenkins/workspace/elastic+logstash+6.x+multijob-unix-compatibility/os/opensuse/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
17:43:03      # /var/lib/jenkins/workspace/elastic+logstash+6.x+multijob-unix-compatibility/os/opensuse/lib/bootstrap/rspec.rb:13:in `<main>'
17:43:03 
17:43:03 Finished in 7 minutes 23 seconds (files took 19.54 seconds to load)
17:43:03 3060 examples, 1 failure, 3 pending
17:43:03 
17:43:03 Failed examples:
17:43:03 
17:43:03 rspec /var/lib/jenkins/workspace/elastic+logstash+6.x+multijob-unix-compatibility/os/opensuse/logstash-core/spec/logstash/agent/converge_spec.rb:105 # LogStash::Agent Agent execute options when `config.reload.automatic` is set to`FALSE` and successfully load the config converge only once
17:43:03 
17:43:03 Randomized with seed 56177
```

This test appears to have had some issues in the past: 
#7737 
#7392

### PR DESCRIPTION
**Pull request by jakelandis**
Fixes #8204 (i think)

Note - I spent along time trying to reproduce the error this is trying to fix, and could not. This is my best guess to root cause.

### CODE REVIEW

File: logstash-core/spec/support/helpers.rb [hunk 1]
```diff
@@ -77,7 +78,9 @@ def start_agent(agent)
     end
   end
 
-  sleep(0.1) unless subject.running?
+  Timeout.timeout(5) do
```
Comment (andrewvc) [inline logstash-core/spec/support/helpers.rb]: I'd prefer we [avoid timeout](http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/).

We should be able to use

`wait(5).for { subject.running? }.to be(true)`

**Comment by jakelandis** [comment]
@andrewvc - thanks!

**Comment by jakelandis** [comment]
Jenkins test this please

**Comment by jakelandis** [comment]
Cancelling this PR in favor of https://github.com/elastic/logstash/pull/8376 (which has this change and others)

**Pull request closed without merging**